sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -63,7 +63,10 @@ case class CreateFunctionCommand(
     } else {
       // For a permanent, we will store the metadata into underlying external catalog.
       // This function will be loaded into the FunctionRegistry when a query uses it.
-      // We do not load it into FunctionRegistry right now.
+      // We do not load it into FunctionRegistry right now. However, we need to validate.
+      catalog.loadFunctionResources(resources)
+      catalog.makeFunctionBuilder(functionName, className)
```

Comment:
Not sure if this is the right direction, but I think we can refactor the whole permanent function resolution. Let me try it.